mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -444,9 +452,27 @@ impl S3CrtClientInner {
                     metrics::histogram!("s3.meta_requests.first_byte_latency_us", latency, "op" => op);
                 }
 
+                // Let the `on_finish` callback decide whether the request failed or not. If it did,
+                // the callback can decide whether it wants to parse the error, or return
+                // `Err(None)` if it wants to fall back to generic error handling.
+                let parsed_result = on_finish(&request_result);
+                let parsed_result = match parsed_result {
+                    Ok(t) => Ok(t),
+                    Err(Some(e)) => Err(Some(e)),
+                    // If `try_parse_generic_error` fails we'll eventually generate a
```

Comment:
This took me a while to digest. Do we use `parsed_result` only for logging? (I mean until line 488 when we are ready to return it)
I wonder if it would get clearer if we moved the logging into the different branches of this `match`.